server/controllers/finance/cash.js

```diff
@@ -152,41 +152,49 @@ function list(req, res, next) {
  * @description list all payment made
  */
 function listPayment(options) {
-  let filters = new FilterParser(options, { tableAlias : 'cash' });
+  let filters = new FilterParser(options, { tableAlias : 'cash_payment' });
 
   let sql = `
-    SELECT BUID(cash.uuid) as uuid, cash.project_id,
-      CONCAT_WS('.', '${identifiers.CASH_PAYMENT.key}', project.abbr, cash.reference) AS reference,
-      cash.date, BUID(cash.debtor_uuid) AS debtor_uuid, cash.currency_id, cash.amount,
-      cash.description, cash.cashbox_id, cash.is_caution, cash.user_id,
-      d.text AS debtor_name, cb.label AS cashbox_label, u.display_name,
-      voucher.type_id, p.display_name AS patientName
-    FROM cash
-      LEFT JOIN voucher ON voucher.reference_uuid = cash.uuid
-      JOIN project ON cash.project_id = project.id
-      JOIN debtor d ON d.uuid = cash.debtor_uuid
-      JOIN patient p on p.debtor_uuid = d.uuid
-      JOIN cash_box cb ON cb.id = cash.cashbox_id
-      JOIN user u ON u.id = cash.user_id
+    SELECT cash_payment.uuid, cash_payment.project_id,
+      cash_payment.reference, cash_payment.date, cash_payment.debtor_uuid, cash_payment.currency_id, 
+      cash_payment.amount, cash_payment.description, cash_payment.cashbox_id, cash_payment.is_caution, 
+      cash_payment.user_id, cash_payment.debtor_name, cash_payment.cashbox_label, cash_payment.display_name,
+      cash_payment.type_id, cash_payment.patientName, cash_payment.patientReference, cash_payment.invoiceReference
+    FROM(
+      SELECT BUID(cash.uuid) as uuid, cash.project_id,
+        CONCAT_WS('.', '${identifiers.CASH_PAYMENT.key}', project.abbr, cash.reference) AS reference,
+        cash.date, BUID(cash.debtor_uuid) AS debtor_uuid, cash.currency_id, cash.amount,
+        cash.description, cash.cashbox_id, cash.is_caution, cash.user_id,
+        d.text AS debtor_name, cb.label AS cashbox_label, u.display_name,
+        voucher.type_id, p.display_name AS patientName, 
+        CONCAT_WS('.', '${identifiers.PATIENT.key}', project.abbr, p.reference) AS patientReference, 
+        do.text AS invoiceReference
+      FROM cash
+        LEFT JOIN voucher ON voucher.reference_uuid = cash.uuid
+        JOIN project ON cash.project_id = project.id
+        JOIN debtor d ON d.uuid = cash.debtor_uuid
+        JOIN patient p on p.debtor_uuid = d.uuid
+        JOIN cash_box cb ON cb.id = cash.cashbox_id
+        JOIN user u ON u.id = cash.user_id
+        LEFT JOIN cash_item ci ON ci.cash_uuid = cash.uuid
```

Comment:
There are a lot of extra JOINs here...  As we saw last week on Friday, this makes the Raspberry Pi very slow.  Are you sure you need these?

Why `LEFT JOIN` on `cash` instead of `JOIN`?